src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -912,6 +907,29 @@ private ProjectInstance CreateTraversalInstance(string wrapperProjectToolsVersio
             return traversalInstance;
         }
 
+        private void EmitMetaproject(ProjectRootElement metaproject, string path)
+        {
+            if (Environment.GetEnvironmentVariable("MSBuildEmitSolution") != null)
```

Comment:
This will get called once per project now instead of once. Ideally we could move this to `Traits` and it would be cached so it doesn't matter.